### PR TITLE
Fix type of Current for TemplatingVariable.CustomVariable.

### PR DESCRIPTION
**Pull request by jsoo1**
I am not 100% sure but I think CustomVariable has the wrong type of `Current` (the values and text should both be lists of text, I think).

### CODE REVIEW

**Comment by weeezes** [comment]
Odd, this is what a quite fresh grafana gives for me for a custom template variable:

```
  "templating": {
    "list": [
      {
        "allValue": null,
        "current": {
          "text": "a",
          "value": "a"
        },
        "description": null,
        "error": null,
        "hide": 0,
        "includeAll": false,
        "label": "Foo",
        "multi": false,
        "name": "foo",
        "options": [
          {
            "text": "a",
            "value": "a"
          },
          {
            "text": "b",
            "value": "b"
          },
          {
            "text": "c",
            "value": "c"
          },
          {
            "text": "d",
            "value": "d"
          },
          {
            "text": "e",
            "value": "e"
          }
        ],
        "query": "a,b,c,d,e",
        "refresh": 0,
        "skipUrlSync": false,
        "type": "custom"
      }
    ]
 }
```

Which version are you seeing this in?

File: defaults/TemplatingVariable.dhall [hunk 1]
```diff
@@ -111,25 +111,24 @@ let mkDatasource =
...
-        ]
-      , query = "6,7,8,9"
-      , skipUrlSync = False
-      , type = VariableType.custom
-      }
+        { allValue = None Text
+        , current = { text = [] : List Text , value = [] : List Text }
+        , hide = 0
+        , includeAll = True
+        , label = None Text
```
Comment (weeezes) [inline defaults/TemplatingVariable.dhall:162]:
```suggestion
        , label = None Text
        , description = None Text
```

Comment (weeezes) [reply defaults/TemplatingVariable.dhall:162]: Made a mistake when "fixing" the merge conflict, sleepy brain

**Comment by weeezes** [comment]
Ah now I think I get it, it's probably multi-value vs single value selection templates? Didn't check this yet, but I'm fairly certain. Hmm, need to be handled in some more involved way I guess :(. Unless grafana actually supports always using lists? But for type definitions it would anyway be nice to have multi vs single value separation clear...

**Comment by jsoo1** [comment]
Oh! I originally made an additional smart constructor for the multi-selection version. I will go back to that.

**Comment by jsoo1** [comment]
Ok, I kept the original CustomVariable and added CustomMultiVariable (for lack of better term). It is really hard to discern what the actual types of these variables are from Grafana's docs...

**Comment by jsoo1** [comment]
Got rid of some of the unwanted changes from `dhall format`.

**Comment by weeezes** [comment]
How about something like this https://github.com/weeezes/dhall-grafana/pull/32 ? Wouldn't need to replicate every single current (and future) templating variable definition, but might be troublesome as users need to set multiple parameters to get it right. Another way would be to somehow create the record types and values from compining some smaller record with current+option fields in either multi or single form and the actual templating variables record.

**Comment by jsoo1** [comment]
I like it. Let me go make my comments there.

**Comment by jsoo1** [comment]
Closing in favor of #32

**Pull request closed without merging**